### ISSUE TITLE
XML output should match the junitreport Ant task

### ISSUE THREAD

**Issue by GoogleCodeExporter** [google/googletest#114]
```
Currently, the XML output mode for the --gtest_output switch produces a single 
XML file 
containing all testsuites and testcase results.  Such a file cannot be 
correctly parsed by some CI 
servers, namely Hudson.  Issue #77 erroneously blames gtest's XML formatting 
for this bug, but 
the actual cause of this problem is that Hudson expects to parse JUnit results 
as one XML file per 
class.

I wrote a perl script to break apart the XML output generated by gtest into 
individual files, and 
called it from ant in hudson.  Doing this fixes the issue, which serves as a 
workaround (I've also 
attached the script to this report).

It would be nice if the gtest_output switch could perhaps include support for 
writing individual 
XML output files.  Perhaps this could be done either with 
"gtest_output:xmldir", or by parsing the 
argument given here to see if it refers to an existing directory (though this 
would probably be 
confusing).

What steps will reproduce the problem?
1. Generate XML output using --gtest_output:xml
2. Configure hudson project to parse JUnit results with this file
3. Hudson then complains that "None of the test reports contained any result"

What version of the product are you using? On what operating system?
1.2.1

```

Original issue reported on code.google.com by `njreiman@gmail.com` on 10 Feb 2009 at 1:45

Attachments:
- [gtest-hudson.pl](https://storage.googleapis.com/google-code-attachments/googletest/issue-114/comment-0/gtest-hudson.pl)

**Comment by gayanpathirage** [comment]
Still not available event in 1.8.0 version; --gtest_output=xml still produce incompatible output. This is required for Jenkins, SonarQube like external analyzers.

mytest --gtest_output=junit_xml:output.xml
WARNING: unrecognized output format "junit_xml" ignored.

**Comment by ramapal** [comment]
It still does not work. The output is incompatible with junit xml parser.
